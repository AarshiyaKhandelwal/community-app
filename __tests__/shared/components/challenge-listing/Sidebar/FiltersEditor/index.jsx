import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import FiltersEditor from 'components/challenge-listing/Sidebar/FiltersEditor';

const changeFilterName = jest.fn();
const deleteSavedFilter = jest.fn();
const dragSavedFilterMove = jest.fn();
const dragSavedFilterStart = jest.fn();
const resetFilterName = jest.fn();
const setEditSavedFiltersMode = jest.fn();
const updateAllSavedFilters = jest.fn();
const updateSavedFilter = jest.fn();

const mockDatas = [
  {
    changeFilterName,
    deleteSavedFilter,
    dragState: {},
    dragSavedFilterMove,
    dragSavedFilterStart,
    resetFilterName,
    savedFilters: [{ id: '1', name: 'name' }],
    setEditSavedFiltersMode,
    updateAllSavedFilters,
    updateSavedFilter,
  },
];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <FiltersEditor {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <FiltersEditor {...this.props} />;
  }
}

test('handle events', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[0]} />));
  let matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('done-button'));
  expect(matches).toHaveLength(1);
  TU.Simulate.click(matches[0]);
  expect(setEditSavedFiltersMode).toHaveBeenCalledWith(false);

  matches = TU.scryRenderedDOMComponentsWithTag(instance, 'input');
  expect(matches).toHaveLength(1);
  TU.Simulate.change(matches[0], { target: { value: 'value' } });
  expect(changeFilterName).toHaveBeenCalledWith(0, 'value');
  expect(resetFilterName).not.toHaveBeenCalled();
  TU.Simulate.keyDown(matches[0], { key: 'Enter' });
  expect(resetFilterName).not.toHaveBeenCalled();
  TU.Simulate.keyDown(matches[0], { key: 'A' });
  expect(resetFilterName).not.toHaveBeenCalled();
  TU.Simulate.keyDown(matches[0], { key: 'Escape' });
  expect(resetFilterName).toHaveBeenCalledWith(0);
  TU.Simulate.blur(matches[0]);
  expect(updateSavedFilter).toHaveBeenCalled();

  matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('right'));
  expect(matches).toHaveLength(1);
  TU.Simulate.click(matches[0]);
  expect(deleteSavedFilter).toHaveBeenCalledWith('1');

  matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('ActiveFilterItem'));
  expect(dragSavedFilterMove).not.toHaveBeenCalled();
  TU.Simulate.drag(matches[0]);
  expect(dragSavedFilterMove).toHaveBeenCalled();
  expect(dragSavedFilterStart).not.toHaveBeenCalled();
  TU.Simulate.dragStart(matches[0]);
  expect(dragSavedFilterStart).toHaveBeenCalled();
});

