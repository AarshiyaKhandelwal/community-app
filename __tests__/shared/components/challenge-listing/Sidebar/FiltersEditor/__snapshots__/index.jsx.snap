// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-challenge-listing-Sidebar-FiltersEditor-___style__EditMyFilters___1O81H"
>
  <h1>
    My filters
  </h1>
  <div
    className="src-shared-components-challenge-listing-Sidebar-FiltersEditor-___style__done-button___3Z310"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    Done
  </div>
  <Item
    changeFilterName={[Function]}
    deleteSavedFilter={[Function]}
    dragSavedFilterMove={[Function]}
    dragSavedFilterStart={[Function]}
    name="name"
    resetFilterName={[Function]}
    updateSavedFilter={[Function]}
  />
  <div
    className="src-shared-components-challenge-listing-Sidebar-FiltersEditor-___style__note___24HgZ"
  >
    Drag the filters to set the order you prefer; use the "x" mark to delete the filter(s) you don't need.
  </div>
</div>
`;
